applications/StructuralMechanicsApplication/custom_strategies/custom_builder_and_solver/explicit_builder_and_solver.hpp

```diff
@@ -124,61 +124,35 @@ class ExplicitBuilderAndSolver
         ElementsArrayType& r_elements       = rModelPart.Elements();
         ProcessInfo& r_current_process_info   = rModelPart.GetProcessInfo();
 
-        #ifdef _OPENMP
-            int number_of_threads = omp_get_max_threads();
-        #else
-            int number_of_threads = 1;
-        #endif
 
-        vector<unsigned int> node_partition;
-        OpenMPUtils::CreatePartition(number_of_threads, r_nodes.size(), node_partition);
-
-        vector<unsigned int> element_partition;
-        OpenMPUtils::CreatePartition(number_of_threads, r_elements.size(), element_partition);
-
-
-        #pragma omp parallel
+        typename NodesArrayType::iterator it_begin=r_nodes.ptr_begin();
+        #pragma omp parallel for 
+        for(size_t i=0;i<r_nodes.size();++i)
         {
+            double& r_nodal_mass    =  it_begin->GetValue(NODAL_MASS);
+            r_nodal_mass = 0.0;
 
-            #pragma omp for
-
-                for(int k=0; k<number_of_threads; k++)
-                {
-                    typename NodesArrayType::iterator i_begin=r_nodes.ptr_begin()+node_partition[k];
-                    typename NodesArrayType::iterator i_end=r_nodes.ptr_begin()+node_partition[k+1];
-
-                    for(ModelPart::NodeIterator i=i_begin; i!= i_end; ++i)
-                    {
-                        double& r_nodal_mass    =  i->GetValue(NODAL_MASS);
-                        r_nodal_mass = 0.0;
-
-                        if (i->HasDofFor(ROTATION_X))
-                        {
-                            array_1d<double,3>& r_nodal_inertia = i->GetValue(NODAL_INERTIA);
-                            noalias(r_nodal_inertia) = ZeroVector(3);
-                        }
-
-                    }
-                }
+            if (it_begin->HasDofFor(ROTATION_X))
+            {
+                array_1d<double,3>& r_nodal_inertia = it_begin->GetValue(NODAL_INERTIA);
+                noalias(r_nodal_inertia) = ZeroVector(3);
+            }
+            it_begin++;
```

Comment:
take care. this is wrong...it_begin is a shared variable, you need to make it firstprivate